skimage/feature/peak.py

```diff
@@ -127,6 +127,8 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     # operating on each label separately
     if labels is not None:
         label_values = np.unique(labels)
+        if np.issubdtype(label_values.dtype, np.bool):
+            label_values = label_values.astype(np.int8)
```

Comment:
I wonder if bool dtype here means they expect a labeling... Is this related to a bug report or failing/warning test?

Reply:
This generates a warning. They implement diff using the subtraction operator so they claim that subtracting book types is no good.

Reply:
Sure but do we know of cases where bool labels have been passed here? it might be safer to just raise on bool

Reply:
It was in a doc test. A binary image was constructed and passed in as a label

Reply:
I imagine it's probably used that way by mistake...

Reply:
Hmm. Ok I'll find the reference and let you decide.